compiler/damlc/lib/DA/Cli/Damlc/Command/MultiIde.hs

```diff
@@ -349,14 +309,24 @@ subIDEMessageHandler miState unblock ide bs = do
             replyLocations :: [LSP.Location] -> IO ()
             replyLocations = reply . Right . LSP.InR . LSP.InL . LSP.List
         case parsedResult of
+          -- Request failed, forwrd error
           Left err -> reply $ Left err
+          -- Request didn't find any location information, forward "nothing"
           Right Nothing -> replyLocations []
+          -- SubIDE containing the reference also contained the definition, so returned no name to lookup
+          -- Simply forward this location
           Right (Just (TryGetDefinitionResult loc Nothing)) -> replyLocations [loc]
+          -- SubIDE containing the reference did not contain the definition, it returns a fake location in .daml and the name
+          -- Send a new request to a new SubIDE to find the source of this name
           Right (Just (TryGetDefinitionResult loc (Just name))) -> do
             debugPrint $ "Got name in result! Backup location is " <> show loc
             let mHome = Map.lookup (tgdnPackageUnitId name) $ multiPackageMapping miState
             case mHome of
+              -- Didn't find a home for this name, we do not know where this is defined, so give back the (known to be wrong)
+              -- .daml data-dependency path
+              -- This is the worst case, we'll later add logic here to unpack and spinup an SubIDE for the read-only dependency
```

Comment:
What happens currently? Is it still unpacking the DAR?

Reply:
it unpacks the DAR elsewhere, the IDE points to the generated data-dep code in package database.